eng/testing/tests.singlefile.targets

```diff
@@ -84,6 +82,21 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="LocateNativeCompiler"
+          Condition="'$(TestNativeAot)' == 'true' and '$(TargetOS)' != 'windows'"
```

Comment:
this should probably use HostOS, not TargetOS

or should we keep the `'$(TargetArchitecture)' != '$(BuildArchitecture)' and '$(HostOS)' == 'linux'` that was there initially?

Reply:
> this should probably use HostOS, not TargetOS

For Windows it doesn't make a difference as we don't support Windows -> Linux or Linux -> Windows cross compilation.

> or should we keep the `'$(TargetArchitecture)' != '$(BuildArchitecture)' and '$(HostOS)' == 'linux'` that was there initially?

That would mean we we only want to run the init-compiler script from this target for cross compilation on Linux, but I wanted to always run this target on all Unix-like OS as we do it for rest of the native components so we can uniformly resolve the exact compiler which we used elsewhere. This will fix issues like https://github.com/dotnet/runtime/pull/77245#issuecomment-1285330856 in non-CI / external-CI use-cases.

Reply:
ok. I'd still prefer $(HostOS) since it makes more sense and it's also what we're using in crossgen2.csproj